### ISSUE TITLE
Loading .obj crashes executable (Windows / QtCreator / CMake 3.4 / MSVC 2017)

### ISSUE THREAD

**Issue by ygrimault** [libigl/libigl#1033]
I created a simple program that asks the user to load a mesh, and everything works fine when loading bunny.off. However, when I try to load armadillo.obj (both from your tuto github https://github.com/libigl/libigl-tutorial-data), it crashes because of this failed assertion:

> Assertion failed: a_startRow >= 0 && blockRows >= 0 && a_startRow <= xpr.rows() - blockRows && a_startCol >= 0 && blockCols >= 0 && a_startCol <= xpr.cols() - blockCols, file c:\...\libigl\external\eigen\eigen\src/Core/Block.h, line 141

My main.cpp is simply:

```#include <igl/readOBJ.h>
#include <igl/readOFF.h>
#include <igl/opengl/glfw/Viewer.h>

#include <iostream>
using namespace std;

Eigen::MatrixXd V;
Eigen::MatrixXd TC;
Eigen::MatrixXd N;
Eigen::MatrixXi F;
Eigen::MatrixXi FTC;
Eigen::MatrixXi FN;

int main(int argc, char *argv[])
{
  // Load a mesh in OFF format
  //igl::readOBJ("../res/objects/armadillo.obj", V, TC, N, F, FTC, FN);
  //igl::readOFF("../res/objects/bunny.off", V, F);

  // Plot the mesh
  igl::opengl::glfw::Viewer viewer;
  //viewer.data().set_mesh(V, F);
  viewer.open_dialog_load_mesh();
  viewer.launch();
}
``` 

Note that it only crashes when using `open_dialog_load_mesh()`, the function `readOBJ()` works on this object. It doesn't on another object which has a material though, and it gives yet another error:

> Assertion failed: (!(RowsAtCompileTime!=Dynamic) || (nbRows==RowsAtCompileTime)) && (!(ColsAtCompileTime!=Dynamic) || (nbCols==ColsAtCompileTime)) && (!(RowsAtCompileTime==Dynamic && MaxRowsAtCompileTime!=Dynamic) || (nbRows<=MaxRowsAtCompileTime)) && (!(ColsAtCompileTime==Dynamic && MaxColsAtCompileTime!=Dynamic) || (nbCols<=MaxColsAtCompileTime)) && nbRows>=0 && nbCols>=0 && "Invalid sizes when resizing a matrix or array.", file c:\...\libigl\external\eigen\eigen\src/Core/PlainObjectBase.h, line 241

I'm using a 64-bit computer with Windows 10. I'm compiling using QtCreator 4.7.1 (Qt version 5.11.2), CMake 3.4 (had to use an earlier version to compile libigl otherwise I had errors when linking iirc), and MSVC 2017 (15.0, x86_amd64). I pulled the latest version of libigl as of 28th November 2018.

Note that I'm also getting a lot of warnings related to conversions that may lead to loss of data, I don't know if it can be linked together or not.

#### Check all that apply (change to `[x]`)
- [x] Windows
- [ ] Mac OS X
- [ ] Linux

**Comment by jdumas** [comment]
Are you using the `dev` branch or the `master` branch? The obj loader in libigl is very simple, so it may not support all flavors of obj. That being said it shouldn't crash. Can you share the other problematic model you mention? Maybe we can add that as a unit test.

**Comment by ygrimault** [comment]
I'm using the `master` branch.
And it does seem that it's indeed a problem with the file I wanted to load, as I managed to load some other files. Here is a list of problems I encountered:

- Objects with a mix of triangular and tet faces would not appear, even though the viewer opens (I haven't tested if the data is correctly loaded or not).
- .obj files having a line like "o XXX.obj" crashed and said that this was the line at fault.
- An object with only vertices and faces (like armadillo.obj) can't be loaded with `open_dialog_load_mesh()`, but they can be with `readOBJ()`.
- An object with information in disorder (like mario_obj2.obj which is attached) can't be loaded with either function. Or at least I think that's the problem with this file

Obj file (had to convert it to txt for Github): [mario_obj2.txt](https://github.com/libigl/libigl/files/2625532/mario_obj2.txt)
Mtl file (had to convert it to txt for Github): [mario_obj2.obj.txt](https://github.com/libigl/libigl/files/2625531/mario_obj2.obj.txt)
Texture: ![mariod](https://user-images.githubusercontent.com/8794945/49162703-bde75d80-f32b-11e8-81eb-082765990aeb.jpg)
